domain/secretbackend/service/service.go

```diff
@@ -556,9 +555,12 @@ func (s *Service) DeleteSecretBackend(ctx context.Context, params DeleteSecretBa
 
 // RotateBackendToken rotates the token for the given secret backend.
 func (s *Service) RotateBackendToken(ctx context.Context, backendID string) error {
-	backendInfo, err := s.st.GetSecretBackend(ctx,
-		secretbackend.BackendIdentifier{ID: backendID},
-	)
+	var backendInfo *secretbackend.SecretBackend
+	err := s.st.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
+		var err error
+		backendInfo, err = s.st.GetSecretBackend(ctx, secretbackend.BackendIdentifier{ID: backendID})
```

Comment:
I can't see the need for `GetSecretBackend` to be run atomic.
It's now called in 2 places, neither of which can be in an atomic operation, since we make calls to the provider, eg for refresh auth and ping. RotateBackendToken is only called by a single worker anyway.